### ISSUE TITLE
Train a wav2vec2.0 base model on my dataset, and the train_accuracy and val_accuracy  look terrible.

### ISSUE THREAD

**Issue by Johumliu** [pytorch/fairseq#3330]
## ❓ Questions and Help

I train a wav2vec2.0 base model on my dataset，valid-percent set 0.05.

At epoch 20 the valid_accuracy: "0.68451", train_accuracy: "0.81603"
[2021-03-09 04:44:18,059][fairseq.trainer][INFO] - begin training epoch 20
[2021-03-09 04:44:18,059][fairseq_cli.train][INFO] - Start iterating over samples
[2021-03-09 04:52:41,134][fairseq_cli.train][INFO] - begin validation on "valid" subset
[2021-03-09 04:53:15,141][valid][INFO] - {"epoch": 20, "valid_loss": "4.602", "valid_ntokens": "6728.91", "valid_nsentences": "26.0111", "valid_prob_perplexity": "602.726", "valid_code_perplexity": "23.827", "valid_temp": "1.979", "valid_loss_0": "4.594", "valid_loss_1": "0.008", "valid_loss_2": "0", "valid_accuracy": "0.68451", "valid_wps": "24608.5", "valid_wpb": "6728.9", "valid_bsz": "26", "valid_num_updates": "2132", "valid_best_loss": "4.602"}
[2021-03-09 04:53:15,143][fairseq.checkpoint_utils][INFO] - Preparing to save checkpoint for epoch 20 @ 2132 updates
[2021-03-09 04:53:15,144][fairseq.trainer][INFO] - Saving checkpoint to checkpoints/checkpoint_best.pt
[2021-03-09 04:53:18,698][fairseq.trainer][INFO] - Finished saving checkpoint to checkpoints/checkpoint_best.pt
[2021-03-09 04:53:20,712][fairseq.checkpoint_utils][INFO] - Saved checkpoint checkpoints/checkpoint_best.pt (epoch 20 @ 2132 updates, score 4.602) (writing took 5.5689730532467365 seconds)
[2021-03-09 04:53:20,713][fairseq_cli.train][INFO] - end of epoch 20 (average epoch stats below)
[2021-03-09 04:53:20,713][train][INFO] - {"epoch": 20, "train_loss": "6.662", "train_ntokens": "108047", "train_nsentences": "417.738", "train_prob_perplexity": "625.681", "train_code_perplexity": "47.601", "train_temp": "1.979", "train_loss_0": "6.658", "train_loss_1": "0.003", "train_loss_2": "0", "train_accuracy": "0.81603", "train_wps": "21299.4", "train_ups": "0.2", "train_wpb": "108047", "train_bsz": "417.7", "train_num_updates": "2132", "train_lr": "0.000458154", "train_gnorm": "0.005", "train_loss_scale": "64", "train_train_wall": "491", "train_gb_free": "23.1", "train_wall": "10968"}

but at epoch 72 the valid_accuracy: "0.92702" is the best, but train_accuracy: "0.00122" is very low.
[2021-03-09 12:15:17,535][fairseq.trainer][INFO] - begin training epoch 70
[2021-03-09 12:15:17,535][fairseq_cli.train][INFO] - Start iterating over samples
[2021-03-09 12:18:26,840][train_inner][INFO] - {"epoch": 70, "update": 69.355, "loss": "6.658", "ntokens": "107885", "nsentences": "417.735", "prob_perplexity": "639.979", "code_perplexity": "517.661", "temp": "1.928", "loss_0": "6.658", "loss_1": "0", "loss_2": "0", "accuracy": "0.00104", "wps": "21240.7", "ups": "0.2", "wpb": "107885", "bsz": "417.7", "num_updates": "7400", "lr": "0.000323077", "gnorm": "0", "loss_scale": "512", "train_wall": "925", "gb_free": "23.1", "wall": "37674"}
[2021-03-09 12:23:43,935][fairseq_cli.train][INFO] - begin validation on "valid" subset
[2021-03-09 12:24:12,892][valid][INFO] - {"epoch": 70, "valid_loss": "1.97", "valid_ntokens": "6749.62", "valid_nsentences": "26.0111", "valid_prob_perplexity": "639.979", "valid_code_perplexity": "3.9", "valid_temp": "1.927", "valid_loss_0": "1.97", "valid_loss_1": "0", "valid_loss_2": "0", "valid_accuracy": "0.92702", "valid_wps": "31136.8", "valid_wpb": "6749.6", "valid_bsz": "26", "valid_num_updates": "7469", "valid_best_loss": "1.97"}
[2021-03-09 12:24:12,893][fairseq.checkpoint_utils][INFO] - Preparing to save checkpoint for epoch 70 @ 7469 updates
[2021-03-09 12:24:12,894][fairseq.trainer][INFO] - Saving checkpoint to checkpoints/checkpoint_best.pt
[2021-03-09 12:24:16,075][fairseq.trainer][INFO] - Finished saving checkpoint to checkpoints/checkpoint_best.pt
[2021-03-09 12:24:18,091][fairseq.checkpoint_utils][INFO] - Saved checkpoint checkpoints/checkpoint_best.pt (epoch 70 @ 7469 updates, score 1.97) (writing took 5.1970200296491385 seconds)
[2021-03-09 12:24:18,092][fairseq_cli.train][INFO] - end of epoch 70 (average epoch stats below)
[2021-03-09 12:24:18,093][train][INFO] - {"epoch": 70, "train_loss": "6.658", "train_ntokens": "107983", "train_nsentences": "417.766", "train_prob_perplexity": "639.979", "train_code_perplexity": "519.122", "train_temp": "1.927", "train_loss_0": "6.658", "train_loss_1": "0", "train_loss_2": "0", "train_accuracy": "0.00122", "train_wps": "21372", "train_ups": "0.2", "train_wpb": "107983", "train_bsz": "417.8", "train_num_updates": "7469", "train_lr": "0.000321308", "train_gnorm": "0", "train_loss_scale": "1024", "train_train_wall": "494", "train_gb_free": "23.1", "train_wall": "38025"}

**Comment by apoca909** [comment]
the dict.ltr.txt size is ?
i meet  similar issue, but valid_wer is keeping 100.
maybe the dict's size matters, but not sure.

**Comment by heibaidaolx123** [comment]
Hi, have you found why?

**Comment by jubick1337** [comment]
@heibaidaolx123 @Nemasery it is common issue if you try to run default config on hardware setup different to 64 V100
You should adjust config to your hardware setup or use ```optimization.update_freq``` to simulate setup.
Otherwise some gradient errors happen during pretraining.
